node/handlerbase.go

```diff
@@ -180,25 +183,48 @@ func (h *HandlerBase) onGetAddr(getAddr *msg.GetAddr) {
 	if h.node.IsExternal() {
 		for _, addr := range LocalNode.RandSelectAddresses() {
 			if addr.Services&protocol.OpenService == protocol.OpenService {
-				addr.Port = config.Parameters.NodeOpenPort
-				addrs = append(addrs, addr)
+				addrs = append(addrs,
```

Comment:
Can change to:
var copyAddr NetAddress = *addr
addrs = append(addrs, &copyAddr)